### PR TITLE
Update DG + Restructuring

### PR DESCRIPTION
**Pull request by mrmrinal**
Added an introduction and Numbering to the DG

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -6,8 +6,31 @@ title: Developer Guide
...
+## 1. Introduction
+
+### 1.1 Purpose
+This document intends to cover the multi-layered design architecture of DonnaFin.io, and can be used by the intended 
+audience in order to better understand the inner workings of the program and better understand the interaction between
+the various components than form DonnaFin.
+
+### 1.2 Target Audience
+This developer guide is aimed at developers and advanced users of DonnaFin
+* Developers: anyone who wish to upgrade Jarvis to support more functions.
```
Comment (bharathcs) [inline docs/DeveloperGuide.md:20]: oops

**Review by bharathcs — CHANGES_REQUESTED** [review]
Please merge upstream/master too!!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ title: Developer Guide
 --------------------------------------------------------------------------------------------------------------------
 <div style="page-break-after: always"></div>
 
-## 1. Introduction
+## **1. Introduction**
```
Comment (bharathcs) [inline docs/DeveloperGuide.md:11]:
```suggestion
## 1. Introduction
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -17,14 +17,19 @@ the various components than form DonnaFin.
 
 ### 1.2 Target Audience
 This developer guide is aimed at developers and advanced users of DonnaFin
-* Developers: anyone who wish to upgrade Jarvis to support more functions.
+* Developers: anyone who wish to upgrade DonnaFin.IO to support more functions.
```
Comment (bharathcs) [inline docs/DeveloperGuide.md:20]:
```suggestion
* Developers: anyone who wish to upgrade DonnaFin to support more functions.
```

**Review by bharathcs — CHANGES_REQUESTED** [review]
Can we not bold the headings? Its a bit inconsistent with the rest of it

**Review by Bluntsord — CHANGES_REQUESTED** [review]
Do u wanna make the deisgn and implementation together right now?

**Review by bharathcs — DISMISSED** [review]
Could we combine implementation & design?

**Review by Bluntsord — APPROVED** [review]

**Pull request merged**